processor/configuration/src/main/java/com/redhat/service/smartevents/processor/resolvers/custom/AnsibleTowerJobTemplateActionResolver.java

```diff
@@ -0,0 +1,52 @@
+package com.redhat.service.smartevents.processor.resolvers.custom;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.enterprise.context.ApplicationScoped;
+
+import com.redhat.service.smartevents.infra.exceptions.definitions.user.GatewayProviderException;
+import com.redhat.service.smartevents.infra.models.gateways.Action;
+import com.redhat.service.smartevents.processor.GatewayResolver;
+import com.redhat.service.smartevents.processor.actions.ansible.AnsibleTowerJobTemplateAction;
+import com.redhat.service.smartevents.processor.actions.webhook.WebhookAction;
+
+@ApplicationScoped
+public class AnsibleTowerJobTemplateActionResolver implements AnsibleTowerJobTemplateAction, GatewayResolver<Action> {
+
+    @Override
+    public Action resolve(Action action, String customerId, String bridgeId, String processorId) {
+        Map<String, String> parameters = new HashMap<>();
+        try {
+            String launchTemplateEndpoint = getLaunchTemplateEndpoint(action);
+            parameters.put(WebhookAction.ENDPOINT_PARAM, launchTemplateEndpoint);
+        } catch (MalformedURLException e) {
+            throw new GatewayProviderException("Can't build API call for ansible tower instance");
+        }
+
+        if (action.hasParameter(BASIC_AUTH_USERNAME_PARAM)) {
+            parameters.put(BASIC_AUTH_USERNAME_PARAM, action.getParameter(BASIC_AUTH_USERNAME_PARAM));
+        }
+        if (action.hasParameter(BASIC_AUTH_PASSWORD_PARAM)) {
+            parameters.put(BASIC_AUTH_PASSWORD_PARAM, action.getParameter(BASIC_AUTH_PASSWORD_PARAM));
+        }
+        if (action.hasParameter(SSL_VERIFICATION_DISABLED)) {
+            parameters.put(SSL_VERIFICATION_DISABLED, action.getParameter(SSL_VERIFICATION_DISABLED));
+        }
+
+        Action transformedAction = new Action();
+        transformedAction.setType(WebhookAction.TYPE);
+        transformedAction.setMapParameters(parameters);
+
+        return transformedAction;
```

Comment:
Side comment, what about the payload? Is it the whole message? Checking Ansible Tower documentation it seems the payload to be an arbitrary JSON so the whole payload should work
https://docs.ansible.com/ansible-tower/latest/html/towerapi/api_ref.html#/

Reply:
The payload is the expected payload from Ansible Tower API call, it's up to the user to transform the original event into a compatible payload. It must be documented in the action reference.